lib/Modules/ComputeHooks/ComputeHooks.psm1

```
@@ -58,6 +58,25 @@ function Install-Prerequisites {
     return $rebootNeeded
 }
 
+function Enable-MPIO {
+    $cfg = Get-JujuCharmConfig
+    if (!$cfg['enable-multipath-io']) {
+        return $false
+    }
+    $mpioState = Get-WindowsOptionalFeature -Online -FeatureName MultiPathIO
+    if ($mpioState.State -like "Enabled") {
+        Write-JujuWarning "MPIO already enabled"
+        $autoClaim = Get-MSDSMAutomaticClaimSettings
+        if (!$autoclaim.iSCSI) {
+            Enable-MSDSMAutomaticClaim -BusType iSCSI -ErrorAction SilentlyContinue
+        }
+        return $false
+    }
+    Write-JujuWarning "Enabling MultiPathIO feature"
+    Enable-WindowsOptionalFeature -Online -FeatureName MultiPathIO -NoRestart -ErrorAction SilentlyContinue
```

The output of this command will tell you if a restart is required. Here you can do something like:

```powershell
$status =  Enable-WindowsOptionalFeature -Online -FeatureName MultiPathIO -NoRestart -ErrorAction SilentlyContinue
return $status.RestartNeeded
```

Also, should we ignore errors here?